### ISSUE TITLE
Switching languages will unselect some of the account / folder filters for the unread messages

### ISSUE THREAD

**Issue by PutinVladimir** [Ximi1970/systray-x#53]
1. I start TB as " LANG=ru_RU thunderbird" go Add-on settings -> Mail tab. Select all 5 imap mail accounts. Save.
2. Restart TB as " LANG=de thunderbird" go Add-on settings -> Mail tab and see 4 accounts partly checked and 1 fully unchecked. Select all checked again.
3. Restart TB as " LANG=ru thunderbird" go Add-on settings -> Mail tab and see 4 accounts partly checked and 1 fully unchecked.

Something broke selected mail accounts when change locale from\to ru_RU\ru.

**Comment by Ximi1970** [comment]
Always thought TB used a hardcoded locale...Need to check out the usage of the LANG var.  
Will also check the preferences behaviour again. Also found another problem in the companion app.

**Comment by PutinVladimir** [comment]
It is uses C library function locale inside as everyone to get locale. And changing env var change function returns.

**Comment by Ximi1970** [comment]
Cannot replicate your problem except for the C-library error. TB does not even change the language using the LANG var. But I can imagine that some internal lib of TB will use the LANG setup and maybe change the used character set and thus changing the "name" of the folder saved with another char set. This will result in unselecting the folder. This is no bug but just how computers work with languages in with different char sets. Cannot do anything about that.

**Comment by PutinVladimir** [comment]
Made some video for you.
https://youtu.be/ctzZ-PzKqgY

**Comment by Ximi1970** [comment]
Thnx for the extra info. You could also check the "Error Console" for the lines  "Stored filters" and "Checked filters" after each language switch.  
The stored filters will be checked for their validity every startup. This is just a textual / character set depended compare of the stored filters and the current accounts and folders I get from TB. If something cannot be matched, the filter entry will be removed.   
Maybe you can figure out why something is being disabled.  

I will add some extra log info to display the data I get from TB. This will make a compare hopefully much clearer.  
Will post here when that is in the develop branch.    

The lines look a bit like this:  
Stored filters: [{"unread":true,"folder":{"accountName":".....  
Checked filters: [{"unread":true,"folder":{"accountName":...  

Please watch out when posting these lines. They WILL contain personal info !

**Comment by Ximi1970** [comment]
Finally got your problem replicated thanks to the video :+1:  And I see what is happening.  
The default folders are translated also but not all of them if your are using a gmail account.  Joy...  

For German the "Inbox" is translated to "Posteingang" and that is not how the filter was saved....

I do not think I can solve this. But I wIll add the extra logging to look if a different "language" independend filtering is possible.

**Comment by PutinVladimir** [comment]
Normal:

```
OS: linux background.js:541:11
Arch: x86-64 background.js:542:11
Nack-Arch: undefined background.js:543:11
Browser: Thunderbird background.js:559:11
Vendor: background.js:560:11
Version: 78.5.1 background.js:561:11
Build: 20201130232704 background.js:562:11
Addon version: 0.6.1 background.js:576:11
Stored filters: [{"unread":true,"folder":{"accountName":"xxxxxxxx@ngs.ru","accountId":"account1","path":"/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"xxxxxxxx@ngs.ru","accountId":"account1","path":"/Отправленные","name":"Отправленные","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"xxxxxxxx@ngs.ru","accountId":"account1","path":"/Входящие","name":"Входящие","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/[Gmail]","name":"[Gmail]","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/[Gmail]/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/Входящие","name":"Входящие","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"zzzzzzzzzzzzz@ya.ru","accountId":"account5","path":"/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"zzzzzzzzzzzzz@ya.ru","accountId":"account5","path":"/Входящие","name":"Входящие","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"vvvv@site.tm","accountId":"account6","path":"/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"vvvv@site.tm","accountId":"account6","path":"/Входящие","name":"Входящие","version":"0.6.1"}}] defaults.js:354:13
Checked filters: [{"unread":true,"folder":{"accountName":"xxxxxxxx@ngs.ru","accountId":"account1","path":"/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"xxxxxxxx@ngs.ru","accountId":"account1","path":"/Отправленные","name":"Отправленные","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"xxxxxxxx@ngs.ru","accountId":"account1","path":"/Входящие","name":"Входящие","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/[Gmail]","name":"[Gmail]","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/[Gmail]/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/Входящие","name":"Входящие","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"zzzzzzzzzzzzz@ya.ru","accountId":"account5","path":"/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"zzzzzzzzzzzzz@ya.ru","accountId":"account5","path":"/Входящие","name":"Входящие","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"vvvv@site.tm","accountId":"account6","path":"/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"vvvv@site.tm","accountId":"account6","path":"/Входящие","name":"Входящие","version":"0.6.1"}}] defaults.js:358:13
```

Switched to de:

```
OS: linux background.js:541:11
Arch: x86-64 background.js:542:11
Nack-Arch: undefined background.js:543:11
Browser: Thunderbird background.js:559:11
Vendor: background.js:560:11
Version: 78.5.1 background.js:561:11
Build: 20201130232704 background.js:562:11
Addon version: 0.6.1 background.js:576:11
Stored filters: [{"unread":true,"folder":{"accountName":"xxxxxxxx@ngs.ru","accountId":"account1","path":"/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"xxxxxxxx@ngs.ru","accountId":"account1","path":"/Отправленные","name":"Отправленные","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"xxxxxxxx@ngs.ru","accountId":"account1","path":"/Входящие","name":"Входящие","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/[Gmail]","name":"[Gmail]","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/[Gmail]/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/Входящие","name":"Входящие","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"zzzzzzzzzzzzz@ya.ru","accountId":"account5","path":"/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"zzzzzzzzzzzzz@ya.ru","accountId":"account5","path":"/Входящие","name":"Входящие","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"vvvv@site.tm","accountId":"account6","path":"/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"vvvv@site.tm","accountId":"account6","path":"/Входящие","name":"Входящие","version":"0.6.1"}}] defaults.js:354:13
Checked filters: [{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/[Gmail]","name":"[Gmail]","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"yyyyyyyy@gmail.com","accountId":"account4","path":"/[Gmail]/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"zzzzzzzzzzzzz@ya.ru","accountId":"account5","path":"/Спам","name":"Спам","version":"0.6.1"}},{"unread":true,"folder":{"accountName":"vvvv@site.tm","accountId":"account6","path":"/Спам","name":"Спам","version":"0.6.1"}}]
```

**Comment by Ximi1970** [comment]
These folders get translated when you switch languages:
Inbox, Trash, Drafts and Outbox.

Problem is that not only the folders marked with de specific types are getting translated but also folders that are not marked with a type get translated. For example, I got somehow a double trash folder from gmail ( /Trash and /{Gmail}/Trash) but only one was marked as trash in the Web extensions API but both got translated. And that messes up the filter check.  

So I added in the feature-filters branch a new filter checker that checks and translates all specially marked folders.  
Looks it works fine for my simple gmail test accounts but I know it is not perfect. And there is no way I can make this perfect sadly because not all data seems to be passed to the web extensions api.  

The extra types checked are: drafts, sent, templates, archives, junk but they are not translated at the moment by TB. If they are also translated in the future they should also automaticaly be translated.
